lib/node_modules/@stdlib/stats/incr/wstdev/test/test.js

```diff
@@ -0,0 +1,208 @@
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var incrwstdev = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof incrwstdev, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns an accumulator function', function test( t ) {
+	t.equal( typeof incrwstdev(), 'function', 'returns a function' );
+	t.end();
+});
+
+tape( 'the function returns an accumulator function (known mean)', function test( t ) {
+	t.equal( typeof incrwstdev( 3.0 ), 'function', 'returns a function' );
+	t.end();
+});
+
+tape( 'the function throws an error if provided a non-numeric value', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			incrwstdev( value );
+		};
+	}
+});
+
+tape( 'the accumulator function ignores zero weights and only updates the weighted sample standard deviation after encountering a non-zero weight', function test( t ) {
+	var acc = incrwstdev();
+	t.equal( acc(), null, 'returns expected value' );
+	t.equal( acc( 2.0, 0.0 ), null, 'returns expected value' );
+	t.equal( acc( 2.0, 2.0 ), 0.0, 'returns expected value' );
+	t.equal( acc( 14.0, 2.0 ), 6.0, 'returns expected value' );
+	t.equal( acc( 25.0, 0.0 ), 6.0, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the accumulator function throws an error if provided a second argument which is not a nonnegative number', function test( t ) {
+	var values;
+	var acc;
+	var i;
+
+	values = [
+		'5',
+		-5.0,
+		NaN,
+		undefined,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			acc = incrwstdev();
+			acc( 2.0, value );
+		};
+	}
+});
+
+tape( 'the accumulator function incrementally computes a weighted sample standard deviation', function test( t ) {
+	var wMeanPrev;
+	var expected;
+	var wVarSum;
+	var actual;
+	var delta;
+	var dataX;
+	var dataW;
+	var xwSum;
+	var wMean;
+	var wSum;
+	var acc;
+	var tol;
+	var N;
+	var x;
+	var w;
+	var i;
+
+	dataX = [ 2.0, 3.0, 2.0, 4.0, 3.0, 4.0 ];
+	dataW = [ 1.0, 2.0, 0.1, 1.8, 9.9, 3.6 ];
+	N = dataX.length;
+
+	acc = incrwstdev();
+
+	wMeanPrev = 0.0;
+	wVarSum = 0.0;
+	xwSum = 0.0;
+	wMean = 0.0;
+	wSum = 0.0;
+	for ( i = 0; i < N; i++ ) {
+		x = dataX[ i ];
+		w = dataW[ i ];
+		xwSum += x * w;
+		wSum += w;
+		wMean = xwSum / wSum;
+		wVarSum += w * (x - wMeanPrev) * (x - wMean);
+		wMeanPrev = wMean;
+		expected = sqrt( wVarSum / wSum );
```

Comment:
Instead of doing it this way, I suggest computing all the expected values manually (e.g., by hand). While this looks correct, prefer computing using, e.g., textbook algorithms from first principles, rather than reimplement another incremental algorithm.